cmd/guacone/cmd/patch.go

```diff
@@ -0,0 +1,397 @@
+		for level := 0; level < len(frontiers); level++ {
+			frontierList := frontiers[level]
+			allNodes := []string{}
+			for _, id := range frontierList {
+				path = append(path, id)
+				allNodes = append(allNodes, id)
+			}
+			fmt.Printf("\n---FRONTIER LEVEL %d---\n", level)
+			poc = append(poc, printNodesInfo(ctx, gqlClient, bfsMap, allNodes)...)
+		}
+
+		fmt.Printf("\n---INFO NODES---\n")
+		if len(infoNodes) == 0 {
+			fmt.Printf("no info nodes found\n")
+		} else {
+			poc = append(poc, printNodesInfo(ctx, gqlClient, bfsMap, infoNodes)...)
+		}
+
+		fmt.Printf("\n---POINTS OF CONTACT---")
+		if len(poc) == 0 {
+			fmt.Printf("\nno POCs found\n")
+		} else {
+			for _, id := range poc {
+				fmt.Printf("\n%s: %s", id, makePOCPretty(bfsMap[id].PointOfContact))
+			}
+		}
+
+		fmt.Printf("\n\n---SUBGRAPH VISUALIZER URL--- \nhttp://localhost:3000/?path=%s\n", strings.Join(removeDuplicateValuesFromPath(path), `,`))
+	},
+}
+
+func printNodesInfo(ctx context.Context, gqlClient graphql.Client, bfsMap map[string]analysis.BfsNode, nodes []string) []string {
+	poc := []string{}
+	for _, id := range nodes {
+		node, err := model.Node(ctx, gqlClient, id)
+
+		if err != nil {
+			fmt.Printf("error: %s \n", err)
+		}
+
+		var pretty string
+		switch node := node.Node.(type) {
+		case *model.NodeNodePackage:
+			if bfsMap[id].Type == analysis.PackageName {
+				pretty = makePkgPretty(*node, false)
+			} else {
+				pretty = makePkgPretty(*node, true)
+			}
+		case *model.NodeNodeSource:
+			pretty = makeSrcPretty(*node)
+		case *model.NodeNodeArtifact:
+			pretty = makeArtifactPretty(*node)
+		}
+
+		fmt.Printf("%s: %s\n", id, pretty)
+
+		if bfsMap[id].PointOfContact.Email != "" {
+			poc = append(poc, id)
+		}
+	}
+	return poc
+}
+
+func makePOCPretty(poc model.AllPointOfContact) string {
+	return fmt.Sprintf("id- %s | email- %s | info- %s | collector- %s | justification- %s | origin- %s | since- %s", poc.Id, poc.Email, poc.Info, poc.Collector, poc.Justification, poc.Origin, poc.Since)
+}
+
+func makePkgPretty(pkg model.NodeNodePackage, version bool) string {
+	pkgString := fmt.Sprintf("pkg:%s/%s/%s", pkg.Type, pkg.Namespaces[0].Namespace, pkg.Namespaces[0].Names[0].Name)
+
+	if version {
+		return fmt.Sprintf("%s@%s", pkgString, pkg.Namespaces[0].Names[0].Versions[0].Version)
+	}
+
+	return pkgString
+}
+
+func makeSrcPretty(src model.NodeNodeSource) string {
+	return fmt.Sprintf("src:%s/%s/%s", src.Type, src.Namespaces[0].Namespace, src.Namespaces[0].Names[0].Name)
+}
+
+func makeArtifactPretty(artifact model.NodeNodeArtifact) string {
+	return fmt.Sprintf("artifact: algorithm-%s | digest:%s", artifact.Algorithm, artifact.Digest)
+}
+
+func getPkgID(ctx context.Context, gqlClient graphql.Client, purl string) (string, error) {
+	pkgInput, err := helpers.PurlToPkg(purl)
+
+	if err != nil {
+		return "", err
+	}
+
+	var pkgFilter model.PkgSpec
+	version := false
+
+	if pkgInput.Version != nil && *pkgInput.Version != "" {
+		pkgFilter = model.PkgSpec{
```

Comment:
what about the `subpath` and `qualifierList`?

Reply:
Well for the purpose of the search we are just dealing with nodes at the packageName and packageVersion level so the sub path and qualifier list aren't necessary to search by.

Reply:
but the `pkgID` will change based on if `subPath` or `qualifierList` is specified.

Reply:
Okay, I changed it.